test/e2e/utils_test.go

```diff
@@ -0,0 +1,77 @@
+//go:build e2e_tests
+// +build e2e_tests
+
+package e2e
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/stretchr/testify/assert"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// namespace create a namespace for a test case, and clean it up after the test finishes.
```

Comment:
```suggestion
// namespace creates a namespace for a test case and cleans it up after the test finishes.
```

Actually, this does not clean this up but it returns a `func` that can clean it up.

What you could do here is use `t.Cleanup()` to register the cleanup like so:

```go
func namespaceForCluster(t *testing.T) *corev1.Namespace {
	namespaceForTestCase, err := clusters.GenerateNamespace(ctx, env.Cluster(), t.Name())
	require.NoError(t, err) // this or return an error but NoError is "equivalent" to exitOnErr(err)

	t.Cleanup(func(){
		assert.NoError(t, clusters.CleanupGeneratedResources(ctx, env.Cluster(), t.Name()))
	})

	return namespaceForTestCase
}
```

Reply:
fixed comments.